api/code/flaskr/model.py

```diff
@@ -64,9 +64,14 @@ def validate_prediction_request(request):
             features.append(float(el))
 
         elif t == 'binary':
-            if not el.isdigit() or int(el) not in vals:
+            try:
+                el_parsed = json.loads(el)
+            except json.decoder.JSONDecodeError:
                 return {'error': {'code': 400, 'msg': f'unallowed value for {field}'}}
-            features.append(int(el))
+                
+            if not isinstance(el_parsed, bool):
+                return {'error': {'code': 400, 'msg': f'unallowed value for {field}'}}
```

Comment:
Please add tests for those lines as well.

Reply:
Done!